src/main/java/core/basesyntax/FileWork.java

```diff
@@ -1,7 +1,39 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
+
 public class FileWork {
+    private static final String REGEX_SPLITTER = "\\W+";
+    private static final String STARTS_WITH = "w";
+    private static final String SPLIT_DELIMITER = " ";
+
     public String[] readFromFile(String fileName) {
-        return null;
+        String[] result = {};
+        StringBuilder filteredWords = new StringBuilder();
+        String[] currentLineArray;
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            String currentLine = bufferedReader.readLine();
+            while (currentLine != null) {
+                currentLineArray = currentLine.toLowerCase().split(REGEX_SPLITTER);
+                for (String currentString : currentLineArray) {
+                    if (currentString.startsWith(STARTS_WITH)) {
+                        filteredWords.append(currentString).append(SPLIT_DELIMITER);
+                    }
+                }
+                currentLine = bufferedReader.readLine();
+            }
+            if (filteredWords.length() == 0) {
+                return result;
+            }
+            result = filteredWords.toString().split(SPLIT_DELIMITER);
+            Arrays.sort(result);
```

Comment:
In future projects let's  move these lines outside of try block, because they can't generate IOException